### PR TITLE
[RISCV] Fold (fmv_x_h/w (load)) to an integer load.

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -16984,6 +16984,17 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
       return Op0.getOperand(0);
     }
 
+    if (ISD::isNormalLoad(Op0.getNode()) && Op0.hasOneUse() &&
```
Comment (preames) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:16987]: Is there something here preventing the load from having an offset?   I think the answer is that the load must be a vector load, and that we don't have base+offset addressing for that form, but could you add an assert here to confirm the offset is zero?

File: llvm/test/CodeGen/RISCV/rvv/fixed-vectors-fp.ll [hunk 2]
```diff
@@ -516,41 +516,33 @@ define void @fabs_v8f16(ptr %x) {
...
-; ZVFHMIN-RV32-NEXT:    fmv.x.h a2, fa5
-; ZVFHMIN-RV32-NEXT:    flh fa5, 12(sp)
-; ZVFHMIN-RV32-NEXT:    and a1, a1, a3
-; ZVFHMIN-RV32-NEXT:    and a2, a2, a3
-; ZVFHMIN-RV32-NEXT:    vmv.v.x v9, a2
-; ZVFHMIN-RV32-NEXT:    fmv.x.h a2, fa5
-; ZVFHMIN-RV32-NEXT:    flh fa5, 14(sp)
+; ZVFHMIN-RV32-NEXT:    lhu a3, 10(sp)
+; ZVFHMIN-RV32-NEXT:    lhu a4, 8(sp)
+; ZVFHMIN-RV32-NEXT:    and a1, a1, a2
```
Comment (preames) [inline llvm/test/CodeGen/RISCV/rvv/fixed-vectors-fp.ll:533]: Off topic, but we might be able to improve the and chains here.
1) With bclri, I think we can avoid the constant materialization.
2) It looks like we do this for every lane, and could possible move that into a vector op.

**Review by preames — APPROVED** [review]
LGTM

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp (hunk 1 above)
Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:16987]: I don't know what the offset field in LoadSDNode is for. We don't use it on RISC-V or X86. The immediate offset is represented by a separate ADD node in the base pointer that we pattern match at isel.

File: llvm/test/CodeGen/RISCV/rvv/fixed-vectors-fp.ll (hunk 2 above)
Comment (topperc) [reply llvm/test/CodeGen/RISCV/rvv/fixed-vectors-fp.ll:533]: I have a patch up to not unroll this. https://github.com/llvm/llvm-project/pull/109232

**Pull request merged**